docs/book/v2/quick-start.md

```diff
@@ -66,99 +71,110 @@ A _RequestHandler_ is a class that receives a request and returns a response, wi
 
-### Request Handlers
+### Middleware
 
 ```php
-namespace Application\Handler;
+namespace Blog\Middleware;
 
-use Application\Entity\Album;
+use Blog\Repository\BlogListRepositoryInterface;
+use Blog\Handler\BlogListnHandler;
+use Fig\Http\Message\StatusCodeInterface;
+use Laminas\Diactoros\StreamFactory;
+use Laminas\Router\RouteMatch;
 use Psr\Http\Message\ResponseFactoryInterface;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
 
-class AlbumDetailMiddleware implements RequestHandlerInterface
+class BlogListMiddleware implements MiddlewareInterface
 {
+    /** @var BlogListRepositoryInterface */
+    public $blogListRepository;
+
     /** @var ResponseFactoryInterface */
-    private $responseFactory;
+    public $responseFactory;
+
+    /** @var BlogListnHandler */
+    protected $handler;
+
     /** @var StreamFactoryInterface */
-    private $streamFactory;
+    public $streamFactory;
 
-    public function __construct(ResponseFactoryInterface $responseFactory, StreamFactoryInterface $streamFactory)
+    public function __construct(BlogListRepositoryInterface $blogListRepository, ResponseFactoryInterface $responseFactory)
     {
-        $this->responseFactory = $responseFactory;
-        $this->streamFactory   = $streamFactory;
+        $this->blogListRepository = $blogListRepository;
+        $this->responseFactory    = $responseFactory;
+        $this->streamFactory      = new StreamFactory();
+        $this->handler            = new BlogListnHandler($this->responseFactory, $this->streamFactory);
     }
 
-    public function handle(ServerRequestInterface $request): ResponseInterface
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        /** @var Album $album */
-        $album = $request->getAttribute('album');
+        /** @var RouteMatch $routeMatch */
+        $routeMatch = $request->getAttribute(RouteMatch::class);
+        $blogId     = $routeMatch->getParam('blog_id');
+        $blogPost   = $this->blogListRepository->findById($blogId);
 
-        $body = $this->streamFactory->createStream('The name of the album is: ' . $album->getName());
-        return $this->responseFactory->createResponse()->withBody($body);
+        // if no album was found, we short-circuit the pipe and return a 404 error:
+        if ($blogPost === null) {
+            return $this->responseFactory->createResponse(
+                StatusCodeInterface::STATUS_NOT_FOUND,
+                sprintf('Blog post with ID %s not found!', $blogId)
+            );
+        }
+
+        // ...otherwise we populate the request with the album and call the RequestHandler
+        $request = $request->withAttribute('blogPost', $blogPost);
+        return $this->handler->handle($request);
```

Comment:
Requiring a user of the `quick-start.md` guide to:

 1. write a middleware
 2. **then** write a request handler that uses `ServerRequestInterface#getAttribute('blogPost')`

... seems a bit much?

Why can't this remain a `RequestHandlerInterface` implementation? :thinking:

Reply:
When I used `RequestHandlerInterface`, it did not forward requests to my written handler and I have to call directly my custom handler

I can run handler and use just middleware just by this way, it's not true?